lib/csv_helper/loader.rb

```diff
@@ -6,26 +6,31 @@ module Loader
       remove_empty_values: true, convert_values_to_numeric: false
     }.freeze
 
-    def load(filename, options = {})
+    def load(file, user, comment = '', options = {})
       delete_all
 
-      records = load_from_csv(filename, options)
+      name = file.is_a?(String) ? file : filename.original_filename
+
+      upload = Upload.new(csv_type: klass.name, user: user, filename: name, comment: comment)
+      records = load_from_csv(file, options)
 
       results = klass.import records[:valid], validate: false, ignore: true
       results.failed_instances = records[:invalid]
+
+      upload.update(ok: true)
       results
     end
 
     private
 
-    def load_from_csv(filename, options)
+    def load_from_csv(file, options)
       records = { valid: [], invalid: [] }
 
       # Since row indexes start at 0 and spreadsheets on line 1,
       # add 1 for the difference in indexes and 1 for the header row itself.
       row_offset = 2 + (options[:skip_lines] || 0)
 
-      SmarterCSV.process(filename, merge_options(options)).each.with_index do |row, i|
+      SmarterCSV.process(file, merge_options(options)).each.with_index do |row, i|
```

Comment:
AR-import is explicitly told not to validate . It only saves the valid records, in the smartercsv loop:
```
  record.errors.any? ? records[:invalid] << record : records[:valid] << record
```

The instruction to use AR-import only uses records[:valid]
```
klass.import records[:valid], validate: false, ignore: true
```